src/ProgressMeter.jl

```diff
@@ -265,8 +265,9 @@ function updateProgress!(p::Progress; showvalues = (), truncate_lines = false, v
         p.desc = desc
     end
     p.offset = offset
+    
     if p.counter >= p.n
-        if p.counter == p.n && p.printed
+        if p.counter == p.n #&& p.printed
```

Comment:
@timholy I'm not sure about the effect of this. What's the purpose of `p.printed`?

Reply:
It's part of ensuring you don't start printing only at finish: you have to print at least one update for this block to print anything at all.